src/simple_graph.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+"""File implements a simple graph data structure."""
+from __future__ import unicode_literals
+
+
+class SimpleGraph(object):
+    """Simple_graph implements a simple graph data structure.
+
+    We support 1 way edges and lots of methods described below.
+
+    Sample structure illustrates the concept.  Nodes a, b, c and d exist
+    and could have any valid python value.  The sub-list contains
+    edges for every node.  For instance, node_a shares an edge with
+    node_b, but node_b has no edge with node_a.  Node_b has no edges
+    of it's own, but node_a, c and b are all neighbors of d.
+
+    Graph:  {
+                "node_a": [(node_b, 1)],
+                "node_b": [],
+                "node_c": [(node_b, 1)],
+                "node_d": [(node_a, 1), (node_b, 1), (node_c, 1)],
+            }
+    """
+
+    def __init__(self):
+        """Initialize an empty simple_graph class instance.
+
+        We don't accept a value here, you must start with an empty graph
+        and add data manually via public methods."""
+        self.graph = {}
+
+    def nodes(self):
+        """Returns a list of all nodes in the graph."""
+        return list(self.graph.keys())
+
+    def edges(self):
+        """Returns a list of all edges in the graph."""
+        edge_list = []
+        for key in self.graph:
+            for n, w in self.graph[key]:
```

Comment:
Thanks for removing the weights from your edge output, at least for your simple graph submission.